nautobot/extras/tests/test_datasources.py

```diff
@@ -794,3 +795,49 @@ def create_empty_repo(path, url, clone_initially=False):
                     clone_initially=False,
                 )
                 MockGitRepo.return_value.diff_remote.assert_called()
+
+    def test_duplicate_repo_url_with_unique_provided_contents(self, MockGitRepo):
+        """Create a duplicate repo but with unique provided_contents."""
+        remote_url = "http://localhost/duplicates.git"
+        repo1 = GitRepository(
+            name="Repo 1",
+            slug="repo-1",
```

Comment:
repo slugs need to be python/graphql safe for `next`:

```suggestion
            slug="repo_1",
```

Same comment below.